lab_2_tokenize_by_bpe/main.py

```diff
@@ -26,6 +38,21 @@ def collect_frequencies(
     :param end_of_word: a token that signifies the end of word
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not (
+        isinstance(text, str) and
+        isinstance(start_of_word, str | None) and
+        isinstance(end_of_word, str)
+    ):
+        return None
+    list_of_tokens = text.split(' ')
+    word_frequencies = {}
+    for word in list_of_tokens:
+        if not prepare_word(word, start_of_word, end_of_word):
+            return None
+        key = prepare_word(word, start_of_word, end_of_word)
```

Comment:
If this was the way to handle mypy, it won't help. Mypy keeps track of variable types, if you've checked output data typing, saved in another variable, it does not automatically mean it will be the same the second time. Well, we know it will, but for formal algorithms like mypy it is not evident, and should not be.